### PR TITLE
Release 3.3.0

### PR DESCRIPTION
**Pull request by FinalAngel**
Improvements & Adaptions

- [x] Adding SCSS Linting #22
- [x] Update docs regarding gulpfile commands
- [x] Adding Jasmin Tests #59 
- [x] Update docs regarding testfile naming conventions

Bonus:
- [x] Documentation revamp
- [ ] Bower Component

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-    "class.js": "~1.0.0",
-    "html5shiv": "~3.7.2",
-    "jquery": "~2.1.1",
-    "outdated-browser": "~1.1.0",
-    "qunit": "~1.17.0",
-    "respond": "~1.4.2",
-    "swfobject": "*",
-    "select2-bootstrap": "~1.3.0"
-  }
+    "name": "aldryn-boilerplate-bootstrap3",
```
Comment (vxsx) [inline bower.json]: did you indent manually?) 
next `bower install --save` is gonna break it most likely

File: CHANGELOG.rst [hunk 2]
```diff
@@ -2,6 +2,13 @@
 Aldryn Boilerplate Bootstrap3
 =============================
 
+3.3.0
+-----
+- added scss-lint
+- moved ``addons/_browser.scss`` and ``addons/_select2.scss`` tp ``libs``
+- changed gulpfile.js file to fit guidelines
+- removed csscomb but keept settings
```
Comment (vxsx) [inline CHANGELOG.rst]: typo

File: CHANGELOG.rst [hunk 3]
```diff
@@ -2,6 +2,13 @@
 Aldryn Boilerplate Bootstrap3
 =============================
 
+3.3.0
+-----
+- added scss-lint
+- moved ``addons/_browser.scss`` and ``addons/_select2.scss`` tp ``libs``
```
Comment (vxsx) [inline CHANGELOG.rst]: typo

File: gulpfile.js [hunk 4]
```diff
@@ -1,40 +1,50 @@
...
-    'sass': './private/sass/'
+    'sass': './private/sass/',
+    'docs': './static/docs'
 };
 
 var patterns = {
-    'images': [paths.images + '*', paths.images + '**/*', '!' + paths.images + '/dummy/*/**'],
-    'js': [paths.js + '*.js', paths.js + '**/*.js', '!' + paths.js + '*.min.js', '!' + paths.js + '**/*.min.js'],
-    'sass': [paths.sass + '*', paths.sass + '**/*']
+    'images': [paths.images + '*', paths.images + '**/*',
```
Comment (vxsx) [inline gulpfile.js]: i would go for separate lines for readability
```
var patterns = {
    images: [
        paths.images + '*',
        paths.images + '**/*',
        //ignore
        '!' + paths.images + 'dummy/*/**'
    ]
}
``` 
etc

File: gulpfile.js [hunk 5]
```diff
@@ -1,40 +1,50 @@
-// IMPORTS
-//##########################################################
+/*!
+ * @author:    Divio AG
+ * @copyright: http://www.divio.ch
+ */
+
 'use strict';
```
Comment (vxsx) [inline gulpfile.js]: I wonder why does jshint not fails on this, global `use strict` should throw an error

**Comment by vxsx** [comment]
usual QQ on quoted properties, but looks very good

File: bower.json (hunk 1 above)
Comment (FinalAngel) [inline bower.json]: yeah let it break it ;)

File: gulpfile.js (hunk 5 above)
Comment (FinalAngel) [inline gulpfile.js]: are you sure? it actually wants me to add use strict

File: docs/contribution/index.rst [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+Everything that is merged to *develop* will be released within the next proper release (1.x.x). Major releases (x.x.x)
+will have their own branches but are always merged agains *develop*.
+
+
+********
+Releases
+********
+
+Always adapt the `CHANGELOG.rst <https://github.com/aldryn/aldryn-boilerplate-bootstrap3/blob/master/CHANGELOG.rst>`_
+and `AUTHORS.rst <https://github.com/aldryn/aldryn-boilerplate-bootstrap3/blob/master/CHANGELOG.rst>`_ accordingly.
```
Comment (darioalbanesi) [inline docs/contribution/index.rst]: wrong link - should be https://github.com/aldryn/aldryn-boilerplate-bootstrap3/blob/master/AUTHORS.rst

File: docs/tips/index.rst [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+When using ``float: left;``, ``display: block;`` is not required anymore as **every** element which is floated gets
+automatically the **block** state. This does not apply for sub-elements.
+
+
+****************
+Hidden Attribute
+****************
+
+With modern HTML5 we can use the ``hidden="hidden"`` attribute which is a **softer** version of ``display: none;``.
```
Comment (darioalbanesi) [inline docs/tips/index.rst]: maybe just mention to check http://caniuse.com/#search=hidden before usage

File: private/sass/addons/_lightbox.scss [hunk 8]
```diff
@@ -1,4 +1,4 @@
-//######################################################################################################################
+// #####################################################################################################################
```
Comment (darioalbanesi) [inline private/sass/addons/_lightbox.scss]: if we don't use it anymore.. why keeping it? :)

File: docs/codestyle/javascript.rst [hunk 9]
```diff
@@ -0,0 +1,1159 @@
...
+- 4 spaces for tabs.
+- Place 1 space before leading curly brace.
+- Place 1 space before the opening parenthesis in ``if``, ``while``, etc.
+- Place 1 space after colon.
+- Place no space before the argument list in function calls and declarations, e.g. ``function fight() { ... }``
+- Set off operators with spaces, e.g. ``var x = 2 + 2;``
+- No whitespace at the end of line or on blank lines.
+- Lines should be no longer than 120 characters. There are 2 exceptions, both allowing the line to exceed 120 characters:
+    - If the line contains a comment with a long URL.
+    - If the line contains a regex literal. This prevents having to use the regex constructor which requires otherwise
```
Comment (ruslanmalogulko) [inline docs/codestyle/javascript.rst]: What about {% blocktrans %} strings? But i'm not sure

Comment (vxsx) [inline docs/codestyle/javascript.rst]: ain't no blocktrans in js

File: docs/codestyle/javascript.rst [hunk 10]
```diff
@@ -0,0 +1,1159 @@
...
+
+Functions
+---------
+
+Function expressions:
+
+.. code-block:: javascript
+
+    // anonymous function expression
+    var anonymous = function () {
```
Comment (ruslanmalogulko) [inline docs/codestyle/javascript.rst]: But it's named also :)

File: docs/codestyle/javascript.rst [hunk 11]
```diff
@@ -0,0 +1,1159 @@
...
+
+    // good
+    var test;
+    if (currentUser) {
+        test = function test() {
+            console.log('Yup.');
+        };
+    }
+
+Never name a parameter arguments. This will take precedence over the arguments object that is given to every function scope.
```
Comment (ruslanmalogulko) [inline docs/codestyle/javascript.rst]: And it's js reserved word also

File: private/sass/addons/_lightbox.scss (hunk 8 above)
Comment (FinalAngel) [inline private/sass/addons/_lightbox.scss]: we still use it: https://github.com/aldryn/aldryn-boilerplate-bootstrap3/blob/develop/docs/guidelines/comments.rst

File: docs/codestyle/javascript.rst (hunk 9 above)
Comment (FinalAngel) [inline docs/codestyle/javascript.rst]: correct :+1:

File: docs/tips/index.rst (hunk 7 above)
Comment (FinalAngel) [inline docs/tips/index.rst]: added a comment

File: private/sass/addons/_lightbox.scss (hunk 8 above)
Comment (darioalbanesi) [inline private/sass/addons/_lightbox.scss]: sorry I meant the lightbox.scss file in general :)

Comment (FinalAngel) [inline private/sass/addons/_lightbox.scss]: yep done :)

**Comment by darioalbanesi** [comment]
lgtm

**Pull request merged**